chips/cc26x2/src/setup/ddi.rs

```diff
@@ -1,76 +1,61 @@
-pub unsafe extern "C" fn ddi16bit_write(
-    ui32Base: u32,
-    ui32Reg: u32,
-    mut ui32Mask: u32,
-    ui32WrData: u32,
-) {
-    let mut ui32RegAddr: u32;
-    let ui32Data: u32;
-    ui32RegAddr = ui32Base
-        .wrapping_add(ui32Reg << 1i32)
-        .wrapping_add(0x400u32);
-    if ui32Mask & 0xffff0000u32 != 0 {
-        ui32RegAddr = ui32RegAddr.wrapping_add(4u32);
-        ui32Mask = ui32Mask >> 16i32;
+pub fn ddi16bit_write(base: u32, reg: u32, mut mask: u32, wr_data: u32) {
```

Comment:
anyway to clean this up? if you the ddi registers are register_bitfield'ed you can probably not use this function. I think I cleaned up ADI in a way that should be similar.

Reply:
Yes there is but not without reworking the setup in mod.rs, which is a beast and out of scope for demo. I can make up other excuses to put this off but that is my only legitimate one.